packages/components/src/DateTimeInput.tsx

```diff
@@ -43,64 +43,65 @@ function removeSeparators(value: string): string {
 
 const EXAMPLES = [addSeparators(DEFAULT_VALUE_STRING)];
 
-const DateTimeInput = React.forwardRef<HTMLInputElement, DateTimeInputProps>(
-  (props: DateTimeInputProps, ref) => {
-    const {
-      className = '',
-      onChange = () => undefined,
-      defaultValue = '',
-      onFocus = () => undefined,
-      onBlur = () => undefined,
-      onSubmit,
-      'data-testid': dataTestId,
-    } = props;
-    const [value, setValue] = useState(
-      defaultValue.length > 0 ? addSeparators(defaultValue) : ''
-    );
-    const [selection, setSelection] = useState<SelectionSegment>();
+export const DateTimeInput = React.forwardRef<
+  HTMLInputElement,
+  DateTimeInputProps
+>((props: DateTimeInputProps, ref) => {
```

Comment:
You shouldn't need to define the props type if using the `forwardRef` generic for it already. It looks like the generic automatically types it as `PropsWithChildren<DateTimeInputProps>`

```suggestion
export const DateTimeInput = React.forwardRef<
  HTMLInputElement,
  DateTimeInputProps
>((props, ref) => {
```